src/Keep.lua

```diff
@@ -419,79 +447,133 @@ local function transformUpdate(keep: Keep, newestData: KeepStruct, release: bool
 		}
 	end
 
-	if corrupted then
-		local replaceData = {
+	if corrupted then -- fix keep if corrupted
+		newestData = {
 			Data = newestData.Data,
 			MetaData = newestData.MetaData or DefaultKeep.MetaData, -- auto locks the session too if new keep
 
 			GlobalUpdates = newestData.GlobalUpdates or DefaultKeep.GlobalUpdates,
 			UserIds = newestData.UserIds or DefaultKeep.UserIds,
 		}
-
-		newestData = replaceData
 	end
 
-	if not isKeepLocked(newestData.MetaData) then
-		newestData.MetaData.ActiveSession = if release and newestData.MetaData.ForceLoad
-			then newestData.MetaData.ForceLoad
-			else DefaultMetaData.ActiveSession
-
+	if not isKeepLocked(newestData.MetaData) then -- keep is available for this server
 		local activeSession = DefaultMetaData.ActiveSession -- give the session to the new keep
-		if release then
-			if newestData.MetaData.ForceLoad then
-				newestData.MetaData.ActiveSession = newestData.MetaData.ForceLoad
-			else
-				activeSession = nil
-			end
 
-			newestData.MetaData.ForceLoad = nil -- remove the force load, if any
+		if isReleasing then
+			activeSession = if newestData.MetaData.ForceLoad then table.clone(newestData.MetaData.ForceLoad) else nil -- switch active session to the new server before releasing session on this server, if any
+			newestData.MetaData.ForceLoad = nil -- remove the ForceLoad, if any
+			newestData.MetaData.LastUpdate = os.time()
+
+			keep.MetaData.ActiveSession = { PlaceID = 0, JobID = "" } -- just in case?
+
+			if activeSession then
+				keep:_release(Promise.resolve(keep)) -- release keep without saving data to prevent servers overwriting each other
+			end
+		else
+			newestData.MetaData.LoadCount = keep.MetaData.LoadCount
 		end
 
 		newestData.MetaData.ActiveSession = activeSession
 
-		newestData.MetaData.LastUpdate = os.time()
-
 		if not empty then
-			keep.LatestKeep = DeepCopy(newestData)
+			keep.LatestKeep = deepCopy(newestData)
+		end
+	else -- keep is not available for this server
+		if keep.MetaData.ForceLoad then -- request different server to release session
+			newestData.MetaData.ForceLoad = table.clone(keep.MetaData.ForceLoad)
+			keep.MetaData.ForceLoad = nil
 		end
 	end
 
 	keep._last_save = os.clock()
-	newestData.MetaData.ForceLoad = keep.MetaData.ForceLoad
-	newestData.MetaData.LoadCount = keep.MetaData.LoadCount
 
 	return newestData, newestData.UserIds
 end
 
-function Keep:_save(newestData: KeepStruct, release: boolean) -- used to internally save, so we can better reveal have :Save()
-	if not self:IsActive() then
+function Keep:_release(updater)
+	if releaseCache[self:Identify()] then -- already releasing
+		return releaseCache[self:Identify()]
+	end
+
+	if self._released then
+		return
+	end
```

Comment:
```
        if releaseCache[self:Identify()] then -- already releasing
		return releaseCache[self:Identify()]
	end

	if self._released then
		return
	end
```

Isn't this redundant since these same checks work made in ```:Release``